### ISSUE TITLE
EndInvoke error with SQL2016 or later SMO

### ISSUE THREAD

**Issue by JanosBerke** [bryan5989/SQLPowerDoc#1]
SQL inventory does not work with newer SMO, SQL 2016+

**Comment by SheepReaper** [comment]
I've encountered a similar issue. I would love to update this, but I won't promise anything will come from me in a reasonable amount of time. I am not the original authors, I just saved this project from Codeplex because i found it useful and didn't want to see it disappear. You are welcome to fork and pull request. I still read my github notifications, I just can't dedicate any time to this project yet. My apologies.

**Comment by SheepReaper** [comment]
@JanosBerke please try the dev-test branch, let me know if it works for you. I'm merging from upstream.

**Comment by JanosBerke** [comment]
Thanks! I will test it and happy to contribute. I have had a conversation with @kendalvandyke too and he promised an updated version as well.

**Comment by larspl** [comment]
Hi Bryan, i solve the issue in my fork :; https://github.com/larspl/SQLPowerDoc

**Comment by SheepReaper** [comment]
Hi @larspl , I've already merged from your fork in #2 , I've put it into a separate branch [dev-test](https://github.com/bryan5989/SQLPowerDoc/tree/dev-test), I was waiting for confirmation that it works from @JanosBerke before merging into master, see #3

**Comment by ThomasVandezande** [comment]
@bryan5989 @larspl ,
Fixed it in my fork. Allthough that orginated from the original code from Codeplex we've maintained inside our company. 
I've tried both your branches but the error kept appearing for me.

In my case the issue is caused by the 'default' statement in the switch for the 'servicetypevalue' in the scriptblock located in the 'networkscan.psm1' file in function 'Find-SqlServerService'. 
In my particular case the 'SQL Server Lanuchpad' type was not defined causing it to jump to the 'default' where the statement '$_.Type.value__' is causing a null-valued exception later on.

So I also updated the 'default' clause to foresee a more decent troubleshooting in the future should Microsoft come up with another SQL Service.

My new code block is the following:
```
					$ServiceTypeName = switch ($_.Type.value__) {
						5 { 'SQL Server Analysis Services' }
						#$ManagedServiceTypeEnum::NotificationServer { 'SQL Server Notification Services' }
						6 { 'SQL Server Reporting Services' }
						#$ManagedServiceTypeEnum::Search { 'Microsoft Search service' }
						3 { 'SQL Server FullText Search' }
						2 { 'SQL Server Agent' }
						7 { 'SQL Server Browser' }
						1 { 'SQL Server' }
						4 { 'SQL Server Integration Services' }
						9 { 'SQL Full-text Filter Daemon Launcher' }
						12 { 'SQL Server Launchpad' }
						$null { 'Unknown' }
						default { $typevalue =  $_.Type.value__
								  Write-NetworkScanLog -Message "Typevalue $Typevalue not found. Please define a value for variable 'ServiceTypeName'." -MessageLevel Information}
					}
```

**Comment by SheepReaper** [comment]
Resolved in #3 
adding @larspl as collaborator. I appear to be distracted at moment.
hopefully the fuller wiki is of some use.
